### ISSUE TITLE
--report-needless-disables should exit with non-zero code 

### ISSUE THREAD

**Issue by evilebottnawi** [stylelint/stylelint#2325]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

```shell
stylelint '{**/*,*}.?(s)css' --ignore-path .gitignore --rd
```

> Which rule, if any, is this issue related to?

No matter

> What CSS is needed to reproduce this issue?

No matter

> What stylelint configuration is needed to reproduce this issue?

No matter

> Which version of stylelint are you using?

latest

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No matter

> What did you expect to happen?

Error code is not equal to 0.

> What actually happened (e.g. what warnings or errors you are getting)?

```
src/components/logo/type-1/_type-1.scss
start: 5, end: 14
```

Error code is equal to 0.

/cc @jeddy3 @davidtheclark

**Comment by evilebottnawi** [comment]
Done in https://github.com/stylelint/stylelint/pull/2341
